challenge-06/qosf-monthly-challenge-06.ipynb

```diff
@@ -0,0 +1,163 @@
+    "\n",
+    "Final board:\n",
+    "<img src=\"images/ex_03.png\" width = 450/>\n",
+    "\n",
+    "Checking the conditions:\n",
+    "\n",
+    "    a. Alice correctly populated her row because her numbers sum up to 0.  \n",
+    "    b. Bob correctly populated her row because her numbers sum up to 0 (good job Bob!).  \n",
+    "    c. Alice and Bob did place the same number in the cell where their row and column intersect: cell (1,2).\n",
+    "    \n",
+    "At last, Alice and Bob have met all conditions and **won** a game.\n",
+    "\n",
+    "### The Challenge\n",
+    "\n",
+    "**Task 1**\n",
+    "Write a program to emulate the game described above following these rules:\n",
+    "\n",
+    "- The rows and column that Richard assigns to Alice and Bob are always selected at random with equal probability.  \n",
+    "- Alice and Bob select the numbers to populate their respective row and column at random with equal probability. \n",
+    "\n",
+    "Run the game several times and extract the probability of winning under each of these conditions.\n",
+    "\n",
+    "(Optional): Calculate the probability of winning under the given conditions to verify the results from your program.\n",
+    "\n",
+    "**Task 2**  \n",
+    "Repeat task 1, but now:\n",
+    "- Alice still picks her numbers to populate her row at random, but she improves her strategy by making sure she only selects numbers that will add up to 0.  \n",
+    "- Bob still picks the numbers to populate his column at random, but he improves his strategy by making sure he only selects numbers that will add up to 1.  \n",
+    "\n",
+    "**Task 3**  \n",
+    "Repeat task 1, but now you utilize a classical strategy to make sure that Alice and Bob very deliberately select how to populate their assigned row and column to maximize their chances of winning the game. Remember, Alice and Bob are not allowed to communicate after their row/column has been assigned.\n",
+    "\n",
+    "Hint: As mentioned in the challenge description, the best possible classical strategy can only guarantee that Alice and Bob win only 8 out of 9 games they play.\n",
+    "\n",
+    "**Task 4**  \n",
+    "Now, to the quantum realm. Alice and Bob understand that one of the rules of the game is that they cannot communicate in any way once the game has started. However, they realize that if they can generated two pairs of entangled qubits, they can build shared correlations that will allow them to win the game every single time (see reference [B] and section 5 of reference [C] for examples of how this could be done). In this task, write a program that emulates the process of generating the two entangled pairs Alice and Bob share, and perform the right unitary operations in each of their qubits (depending on the row/column they are assigned) so that when they perform a measurement, they can generate the right string of ones and zeros to fill out their row/column to win each game.\n",
```

Comment:
```suggestion
    "Now, to the quantum realm. Alice and Bob understand that one of the rules of the game is that they cannot communicate in any way once the game has started. However, they realize that if they can generate two pairs of entangled qubits, they can build shared correlations that will allow them to win the game every single time (see reference [B] and section 5 of reference [C] for examples of how this can be done). In this task, write a program that emulates the process of generating the two entangled pairs Alice and Bob share, and perform the right unitary operations in each of their qubits (depending on the row/column they are assigned) so that when they perform a measurement, they can generate the right string of ones and zeros to fill out their row/column to win each game.\n",
```